Feat: FMBench

jxw-aws opened the pull request:
Latency and throughput testing for the model endpoints by running [FMBench](https://github.com/aws-samples/foundation-model-benchmarking-tool/).

jxw-aws committed:
Latency and throughput testing for the model endpoints by running FMBench.

Co-authored-by: Alex Thewsey <thewsey@amazon.com>

athewsey commented on infra/cdk_src/perf_test/functions/handler.py:
```diff
@@ -0,0 +1,89 @@
+    # delete all files in S3 bucket
+    bucket_name = physical_id[len("fmbench-setup-") :]
+    s3 = boto3.client("s3")
+    #  delete all objects in the bucket
+    # get list_objects paginator
+    response = s3.get_paginator("list_objects_v2").paginate(Bucket=bucket_name).build_full_result()
+    if "Contents" in response:
+        for obj in response["Contents"]:
+            s3.delete_object(Bucket=bucket_name, Key=obj["Key"])
```
This feels kind of risky as a general pattern (CDK Bucket construct can already provide an option to clear bucket on delete), but we don't use this bucket for anything else in the solution so it should be fine for now

athewsey commented on infra/cdk_src/perf_test/functions/handler.py:
```diff
@@ -0,0 +1,89 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+"""CDK Custom Resource Lambda to copy content into an Amazon S3 Bucket"""
```
This file could probably be named something more descriptive than handler.py, but not important

athewsey commented on infra/cdk_src/perf_test/pipeline/assets/experiments/llama3/config.yaml:
```diff
@@ -0,0 +1,237 @@
+general:
+  name: "llmevalworkshop"
+  model_name: "llama-mistral"
```
`llama-mistral`? Could this be `llmevalworkshop-llama` and `llama3`, or would that break things I wonder?

athewsey reviewed (approved):
Some minor notes, but most can be ignored and the one about config.yaml I'm already half way through testing a change for: So no action needed, I'll merge once it's in.

Thanks for all your hard work on this!

athewsey committed:
doc(fmbench): Minor tidy-ups

Add clean-up instructions on base README. Update experiment name and
top-level model name in Llama3 FMBench config yaml. Fix old
`LLMEValWKshpStack` typo spotted in README. Add header & tweak a
couple of comments in FMBench Dockerfile.
